org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateAdaptorApplication.mtl

```diff
@@ -168,9 +169,14 @@ public class [javaClassNameForAdaptorApplication(anAdaptorInterface) /] extends
            throws OslcCoreApplicationException,
                   URISyntaxException
     {
-        super(RESOURCE_CLASSES,
-              OslcConstants.PATH_RESOURCE_SHAPES,
-              RESOURCE_SHAPE_PATH_TO_RESOURCE_CLASS_MAP);
+        registerClasses(RESOURCE_CLASSES);
+        //TODO: As was done in the Wink application, we want to verify that all OSLC4J resource classes are well annotated.
```

Comment:
This current code works fine, and matches what was done earlier.
But I wonder if it is better to move this code to the servletListener.

Reply:
Are we talking about the ResourceShapeFactory use here? I think we should move it but not convinced the `ServletListener` is really the right class.

Reply:
yes, I am talking about the call to ResourceShapeFactory.
it is something we wnat to do at the start of the application. I can't think of any other place than Listener, or Application.